kirill-konshin/next-redux-wrapper
# [BUG]: useMemo Being a rouge agent on useHybridHydrate

##  What happened ? 
While attempting to utilize the wrapper store within the getServerSideProps function provided by the wrapper, an error was encountered. Specifically, this error was initially observed when transitioning from statically generated pages (SSG) to server-side rendered (SSR) pages. However, the issue was further compounded when attempting to revisit the same SSR link from the navigation bar, causing the error to appear twice consecutively.

<img width="1728" alt="Screenshot 2023-04-17 at 3 05 48 AM" src="https://user-images.githubusercontent.com/25548490/232364107-53c6626c-8dd6-4994-b04f-6a830dd62c47.png">

## What did i change ? 
I have implemented a fundamental change in how we handle hydration updates. Although the code was already impressive, the warning message made it clear that a setState call was being triggered within the render method of the App component while a different component (PlaylistView) was being rendered. This action is prohibited in React as it results in unpredictable behavior and may cause the application to crash or malfunction.

To tackle the issue I update the check mechanism. By allowing React to complete its lifecycle, proper data handling is ensured, resulting in an overall improvement in the fluidity of the data hydration process.

## How can I reproduce this error? 
- Run a dispatch from your getServerSideProps that in turn updates the HYDRATE state
- Try to navigate fluidly on your application from CSR -> SSG -> SSR and this bug should be reproduced 

## Reviewers Note
This entire issue is caused by an early` useMemo` what we want to do is hydrate when you are unMounting and rehydrate when you are Mounting "easy peezy"
